File: 1704 C. Virus/Solution.py
# 314198807	      Apr/06/2025 11:38UTC+6	      Krishno_MD	      1704C - Virus	      PyPy 3-64	      Accepted	      265 ms	      18200 KB

#Krishno_MD_KUET_CSE

t = int(input())

for i in range(t):
    n,m = map(int,input().split())
    inf = list(map(int,input().split()))
    inf.sort()
    gaps = []
    for i in range(0,m):
        if i==m-1:
            gaps.append(n-inf[i]+inf[0]-1)
        else:
            gaps.append(inf[i+1]-inf[i]-1)
    gaps.sort(reverse = True)
    protect = 0
    for i in range(0,len(gaps)):
        gaps[i] -= i*4
        if gaps[i]==1:
            protect += 1
        elif gaps[i]>1:
            protect += (gaps[i]-1)
        else:
            break
    print(n-protect)

File: 1704 C. Virus/Solution.cpp
// 314198738	      Apr/06/2025 11:37UTC+6	      Krishno_MD	      1704C - Virus	      C++20 (GCC 13-64)	      Accepted	      203 ms	      100 KB

#include<bits/stdc++.h>
using namespace std;

//Krishno_MD_KUET_CSE

void solution(){
	int n,m;
    cin>>n>>m;
    vector<int> infected(m);
    for(int i=0; i<m; ++i){cin>>infected[i];}
    sort(infected.begin(),infected.end());
    vector<int> gaps;
    for(int i=0; i<m; i++){
        if(i==m-1){
            gaps.push_back(n-infected[i]+infected[0]-1);
        }else{
            int c = infected[i];
            int next = infected[i+1];
            gaps.push_back(next-c-1);
        }
        
    }
    sort(gaps.rbegin(),gaps.rend());
    
    int protected_count = 0;
    for(int i=0; i<gaps.size(); i++){
        gaps[i] -= i*4;
        if(gaps[i]==1){
            protected_count++;
        }else if(gaps[i]>1){
            protected_count += (gaps[i]-1);
        }else{
            break;
        }
    }
    cout<<n - protected_count<<endl;
}

int main(){
	int t; cin>>t;
	for(int i=0; i<t; i++){solution();}
	return 0;
}
